### ISSUE TITLE
[Bug]: Custom login ui / v2 APIs - Error when login with Apple

### ISSUE THREAD

**Issue by quochuydev** [zitadel/zitadel#8892]
### Preflight Checklist

- [x] I could not find a solution in the documentation, the existing issues or discussions
- [x] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment - Both cloud and self-host

ZITADEL Cloud 2.62.8

ZITADEL self-host 2.60.x

### Describe the problem caused by this bug

I set up an Apple identity provider, it login successfully in Zitadel Login UI, but when I use the V2 Intents APIs, it redirects to failed URL:

```
/login/idp/fail?error=oauth2%3A+"invalid_client"&error_description=&id=292706142981894472
```

I guess the error when handling in router `/idps/callback`

https://zitadel.com/docs/apis/resources/user_service_v2/user-service-start-identity-provider-intent

**Comment by livio-a** [comment]
@quochuydev did you set up the callback url for the intents as well?

**Comment by quochuydev** [comment]
@livio-a yes, I already tested with a callback to `login-url/idps/callback` and `zitadel-url/idps/callback`. It returns the same error.

**Comment by livio-a** [comment]
@peintnermax did you have similar / any issues when implementing Apple for typescript repository?

**Comment by quochuydev** [comment]
hi @peintnermax Looking forward to hearing back from you—any news?
cc @livio-a @hifabienne

**Comment by peintnermax** [comment]
There seems to be something odd with your configuration. https://github.com/zitadel/typescript showcases a working example of the apple idp. You can clone the repository and set it up on your instance. https://zitadel.com/docs/guides/integrate/login-ui/typescript-repo
Did you start the flow with this request? https://zitadel.com/docs/apis/resources/user_service_v2/user-service-start-identity-provider-intent

**Comment by quochuydev** [comment]
```
/idp/apple/failure?error=oauth2%3A+%22invalid_client%22&error_description=&id=297326794292362702&organization=233613364822053113
```

@peintnermax, I got the same error when using `zitadel/typescript `

**Comment by quochuydev** [comment]
hi @peintnermax Looking forward to hearing back from you—any news?
cc @livio-a @hifabienne

**Comment by hifabienne** [comment]
Hi @quochuydev 
We have tested it several times, and the typescript repo works for us. So the implementation in general should be fine.

@peintnermax asked you some questions above. Which you should answer so we can help you.
Can you also tell us how your configuration looks like. In ZITADEL as well as in Apple?

Which guides did you follow for that integration?

**Comment by quochuydev** [comment]
hi @hifabienne 
Following this document https://zitadel.com/docs/guides/integrate/identity-providers/apple

This Worked in cloud with Zitadel login UI, but when I used the API /v2/idp_intents to create the `authURL`, the request was handled in `/idps/callback` then redirected to the failed URL.

I got the same error when I implemented it in our project and tested it with your typescript repository.

Do you have the link for the demo? The typescript Zitadel didn't even check Apple implementation
https://github.com/zitadel/typescript?tab=readme-ov-file#features-list

**Comment by hifabienne** [comment]
The error message shows something like invalid client, did you add the correct client id in zitadel? the correct team id and the correct key id?

**Comment by quochuydev** [comment]
> The error message shows something like invalid client, did you add the correct client id in zitadel? the correct team id and the correct key id?

Yes, it's correct, it worked with Zitadel login UI, so adding a login URL to the whitelist Url, it should work.

**Comment by quochuydev** [comment]
Hi @hifabienne can I test login with Apple in your Vercel URL (https://typescript-login.vercel.app)
I tried clone the repo and do local testing but it's not worked

**Comment by quochuydev** [comment]
Hi @hifabienne can I test login with Apple in your Vercel URL (https://typescript-login.vercel.app/)
I tried cloning the repo and doing local testing, connecting to my cloud, but it's not worked

**Comment by hifabienne** [comment]
Hi @quochuydev 
Sorry I can't give you access to my environment.
You can send us screenshots from your exact configuration, in apple and Zitadel so we can compare what the difference could be

**Comment by quochuydev** [comment]
Hi @hifabienne 
Could you have a demo of the login flow with Apple in your custom login UI?

**Comment by livio-a** [comment]
I'm closing this. @quochuydev if you still have the issue, please reopen it and send us the configuration from Apple and Zitadel.

**Comment by spacemule** [comment]
Not sure if I should comment here or open a new issue, but I've seen the exact same behavior. It seems if the IDP was edited, then login v2 fails. For example, set it up with terraform, then replace the secret material after the IDP is already created.

Deleting the IDP and recreating fixes this, but breaks existing linked users.
